classes/utils.php

```diff
@@ -128,4 +130,29 @@ public static function reindex_integer_arrays(array &$array): void {
             $array = array_values($array);
         }
     }
+
+    /**
+     * Returns a list of relevant context ids related to the given context.
+     *
+     * If the given context is part of a course context, the course context id and every child context id are returned.
+     * Else, only the id of the given context is returned inside the array.
+     *
+     * @param context $context
+     * @return int[] relevant context ids
+     * @throws moodle_exception
+     */
+    public static function get_relevant_context_ids(context $context): array {
+        // If context is part of a course, get every context of that course.
+        $coursecontext = $context->get_course_context(false);
+        if ($coursecontext) {
+            // Context is part of a course.
+            $contexts = $coursecontext->get_child_contexts();
+            $contextids = array_keys($contexts);
+            $contextids[] = $coursecontext->id;
```

Comment:
Es gibt einen besseren Weg, DB-Anfragen nach child-Kontexten zu filtern. Schau mal in die Funktion `\context::get_child_contexts`. Da werden die child-Kontexte anhand des Pfades mit einem LIKE-query selektiert. So kannst du das in deinem Query auch machen, dann sparst du dir eine DB-Anfrage.

Reply:
Hier kann man sich noch anschauen, ob das auch beim `search_packages`-Service eine bessere Perfomance liefert.